kagome_lattice.py

```diff
@@ -0,0 +1,190 @@
+
+    # nearest neighbour positions
+    n_pos_a = [[i + ne_a] for i in a]
+    n_pos_b = [[i + ne_b] for i in b]
+    n_pos_c = [[i + ne_c] for i in c]
+
+    # for each element in sites sites_pos gives its nearest neighbours
+    site_pos = []
+    for i in range(size):
+        site_pos.append(n_pos_a[i])
+        site_pos.append(n_pos_b[i])
+        site_pos.append(n_pos_c[i])
+
+    return site_pos
+
+
+def _finding_index(boundary):
+    ind = []
+    for i in boundary:
+        # finding the index of the boundary sites in sites[]
+        us = list(np.where(np.round(sites, 5) == np.round(i, 5)))[0]
+        # print(us)
+        res = [idx for idx, val in enumerate(us) if val in us[:idx]]
+        # print(res)
+        if res != []:
+            ind.append(us[res[0]])
+    return ind
+
+
+def adjacencyList(site_pos, sites, boundary_ind_a, boundary_ind_b, boundary_ind_c, periodic):
+
+    adjacency_list = []
+    v1 = v2 = v3 = v4 = v5 = v6 = 0
+    for i in range(len(site_pos)):
+        ind = []
+        for j in range(len(site_pos[i][0])):
+            # finding the index of the neighbour sites in sites[]
+            us = list(np.where(np.round(sites, 5) ==
+                               np.round(site_pos[i][0][j], 5)))[0]
+            # print(us)
+            res = [idx for idx, val in enumerate(us) if val in us[:idx]]
+            # print(res)
+            if res != []:
+                ind.append(us[res[0]])
+
+        if periodic == 1:
+            if len(ind) != 4:
+                if i in boundary_ind_b:
+                    if i < (Lx*3*(Ly-1)+1):
+                        ind.append(boundary_ind_c[Ly+v1])
+                        v1 = v1+1
+                    if i >= (Lx*3*(Ly-1)+1):
+                        ind.append(boundary_ind_a[Lx+v2])
+                        ind.append(boundary_ind_c[Ly+v1])
+                        v2 = v2+1
+                elif i in boundary_ind_a:
+                    if i <= (Ly*3-1):
+                        ind.append(boundary_ind_b[Ly+v3])
+                        v3 = v3+1
+                    if (i > (Ly*3-1) or i == 0):
+                        ind.append(boundary_ind_c[Lx+v4])
+                        v4 = v4+1
+                else:
+                    if i < (Ly*3-1):
+                        ind.append(boundary_ind_b[Ly+1+v5])
+                        v5 = v5+1
+                    else:
+                        ind.append(boundary_ind_a[v6])
+                        ind.append(boundary_ind_b[v6])
+                        v6 = v6+1
+            ind.sort()
+            # if len(ind) !=4:
+            #     print('bad')
+            # else:
+            #     print(i,ind)
+
+        for k in ind:
+            if [i, k] and [k, i] not in adjacency_list:  # removing symmetery
+                adjacency_list.append([i, k])
+
+    adjacency_list = np.array(adjacency_list)
+    return adjacency_list
+
+
+def make_kagome(Lx, Ly, periodic, name, comment):
+    global Lx, Ly, size, num_sites
```

Comment:
This doesn't work. You can't have these variables both as function arguments and globals. You'll have to pass them as arguments to the functions that need them.

Reply:
Alright. I will pass them as function arguments